arangod/Cluster/ClusterMethods.cpp

```diff
@@ -3750,10 +3750,17 @@ arangodb::Result hotBackupCoordinator(ClusterFeature& feature, VPackSlice const
       return result;
     }
 
+    auto releaseAgencyLock = scopeGuard([&]{
+      LOG_TOPIC("52416", INFO, Logger::BACKUP)
+          << "Releasing agency lock with scope guard! backupId: " << backupId;
+      ci.agencyHotBackupUnlock(backupId, timeout, supervisionOff);
+    });
```

Comment:
If you actually assume this cannot happen, should we add a `TRI_ASSERT(false)` here, so it might help us understand why it does happen anyway sometimes?